src/bin/pg_dump/pg_dump.c

```diff
@@ -2155,9 +2160,23 @@ dumpTableData_insert(Archive *fout, const void *dcontext)
 			appendPQExpBufferStr(q, "NULL");
 		else
 			appendPQExpBufferStr(q, fmtId(tbinfo->attnames[i]));
+		if (is_babelfish_authid_login_ext && pg_strcasecmp(tbinfo->attnames[i], "orig_loginname") == 0)
+			found = true;
 		attgenerated[nfields] = tbinfo->attgenerated[i];
 		nfields++;
 	}
+
+	/*
```

Comment:
let put this logic in a separate bff function if possible. It is fine if we have to iterator over column list again. Also, first using rolname and later replacing orig_loginname, can we do "select rolename as orig_loginname"?